### ISSUE TITLE
awful anti-aliasing and hinting, and no sub-pixel rendering; on linux

### ISSUE THREAD

**Issue by bdjnk** [atom/atom#17007]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Atom does an inexcusably terrible job of anti-aliasing and hinting, and doesn't use sub-pixel rendering.

![atom-garbage-font-rendering](https://user-images.githubusercontent.com/620722/37737758-22751a5c-2d2b-11e8-9ac5-b10b0dd0d99a.png)

The issue is present at any font size, but a small font size, as the screenshot above, makes the problem obvious.

No other application on this system does this, including other Electrons applications, like Slack and Discord.

### Steps to Reproduce

1. Have an up to date install of Arch Linux.
2. Install Atom.
3. Open any file containing text.
4. Reduce the font size to make the effect more noticable.

**Expected behavior:** Decent anti-aliasing, hinting, and sub-pixel rendering.

**Actual behavior:** Garbage.

**Reproduces how often:** Always.

### Versions

atom --version
```
Atom    : 1.24.0
Electron: 1.8.4
Chrome  : 59.0.3071.115
Node    : 8.2.1
```

apm --version
```
apm  1.19.0
npm  5.7.1
node 9.9.0 x64
atom 1.24.0
python 2.7.14
git  2.16.2
```

uname -a
```
Linux PowerMAP 4.15.4-1-ARCH #1 SMP PREEMPT Sat Feb 17 16:01:38 UTC 2018 x86_64 GNU/Linux
```

### Additional Information

I don't know which additional Linux package version might be relevant. If you need more information, I'd be happy to provide it.

**Comment by Arcanemagus** [comment]
Thanks for reaching out!

We've determined that this issue is being caused by the use of an unofficial build or distribution of Atom. Often these customized versions of Atom are modified versions of the Stable branch of Atom with mismatched versions of built-in components. These updated components are taken from the Beta channel or `master` branch and then injected into the Stable version and a new Atom package is generated. Because of the way Atom is constructed, using these mismatched components can cause mysterious and hard-to-diagnose problems. You can find out more about [why we chose to not support unofficial distributions here](https://github.com/atom/design-decisions/blob/master/unofficial-atom-distributions.md).

**Comment by bdjnk** [comment]
Hi @Arcanemagus. Thanks for pointing out my error!

I've determined that that the official distribution of Atom has the exact same problem.

![atom-official-garbage-font-rendering](https://user-images.githubusercontent.com/620722/37739949-40d7d532-2d32-11e8-983d-23a42fbac96d.png)

atom --version
```
Atom    : 1.25.0
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

apm --version
```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.25.0
python 2.7.14
git 2.16.2
```

**Comment by bdjnk** [comment]
If this bug is now tainted with unofficialness I'd be happy to create a new bug based on the official Atom distribution.

**Comment by Arcanemagus** [comment]
Thanks for trying things out with an official build 👍

Are you using any sort of display scaling?

**Comment by bdjnk** [comment]
I'm not using any sort of display scaling.

xrandr | grep -A 1 ' connected'
```
DVI-I-1 connected 1920x1200+1920+0 (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
--
DVI-D-0 connected 1920x1200+0+0 (normal left inverted right x axis y axis) 518mm x 324mm
   1920x1200     59.95*+
```

I discovered something of note. While even a fresh install of Atom has this problem `atom --safe` doesn't.

**Comment by bdjnk** [comment]
Additional possibly relevant screen information.

xdpyinfo | grep -B2 resolution
```
screen #0:
  dimensions:    3840x1200 pixels (1049x321 millimeters)
  resolution:    93x95 dots per inch
```

**Comment by ozra** [comment]
I will comment here, and in other issues I find matching, in an attempt to provide some insight:

- Font rendering has been blurry in Atom since I started using it, probably 2015, possibly earlier.
- Expectation: be a clear as Sublime, or any other program
- Can confirm I've also tried all the kinds of style-tricks as shown above, multiple times, every few months I go through trying things out again. No change with any update so far :-(.
- Have also tried all tips for command-line parameters I've come across from different forums, since 2015-ish, no change
- OS, from initially Ubuntu 15-ish, to currently 18.10
- machines have been ASUS with lofi GFX to currently ASUS with Nvidia GTX 1060  
- Same blurry anti-alias through all that
- Optimally it should be possible to render raster-fonts _without_ anti-alias, for my taste, but _any_ improvement would make my day

Crystal clear font rendering is _fundamental_ for a tool which primary purpose is interacting with text!

**We all have one pair of eyes! _The font-rendering issue really should have top priority imho_.**

**Comment by lee-dohm** [comment]
@ozra Thanks for your feedback. Unfortunately, because we use the Chromium engine for rendering, we don't directly control the font rendering system. There are some things we can do to ensure that the Chromium engine uses its idea of the optimum font rendering for the platform but beyond that we are at the mercy of the engine we've selected.

Given the above, the piece of information that you can supply to us that you haven't is if you experience the same problem using the same Chromium version on your machine and OS as Atom employs. If you would please open a new issue and fill out the entire template, we can see if we can reproduce the problem.

Thanks.

**Comment by ozra** [comment]
Quick google gave: `Google has finally released an update -version number 69.0.3497.92 – for Chrome 69.0.3497.81 that includes a fix for blurry fonts on Windows, Mac, and Linux platforms.`. Looking in "about Atom", I see Chome version 61.0.3163.100. So maybe there's hope?

**Comment by lee-dohm** [comment]
> Looking in "about Atom", I see Chome version 61.0.3163.100. So maybe there's hope?

It may be a while before Chromium 69 percolates down to Atom. Electron has to adopt the Chromium engine version first. Then Atom needs to adopt the version of Electron that uses that Chromium engine. But yes, if Chromium 69 is a fix for the issue you're experiencing, then it will reach Atom eventually.
